examples/triangle/main.rs

```diff
@@ -43,26 +49,68 @@ const TRIANGLE: [Vertex; 3] = [
 const CLEAR_COLOR: [f32; 4] = [0.1, 0.2, 0.3, 1.0];
 
 pub fn main() {
+    // Create window
     let events_loop = glutin::EventsLoop::new();
     let builder = glutin::WindowBuilder::new()
         .with_title("Triangle example".to_string())
         .with_dimensions(1024, 768)
         .with_vsync();
-    let (window, mut device, mut factory, main_color, mut main_depth) =
-        gfx_window_glutin::init::<ColorFormat, DepthFormat>(builder, &events_loop);
-    let mut encoder: gfx::Encoder<_, _> = factory.create_command_buffer().into();
-    let pso = factory.create_pipeline_simple(
+    let window = gfx_window_glutin::build(builder, &events_loop, ColorFormat::get_format(), DepthFormat::get_format());
+
+    // Acquire surface and adapters
+    let (mut surface, adapters) = gfx_window_glutin::Window(&window).get_surface_and_adapters();
+    let queue_descs = adapters[0].get_queue_families().iter()
+                                 .filter(|family| surface.supports_queue(&family) )
+                                 .map(|family| { (family, 1) })
+                                 .collect::<Vec<_>>();
+
+    // Open device (factory and queues)
+    let gfx::Device { mut factory, mut general_queues, mut graphics_queues, .. } = adapters[0].open(&queue_descs);
+    let mut graphics_queue = if let Some(queue) = general_queues.first_mut() {
+        queue.as_mut().into()
+    } else if let Some(queue) = graphics_queues.first_mut() {
+        queue.as_mut()
+    } else {
+        panic!("Unable to find a matching general or graphics queue.");
+    };
+
+    // Create swapchain
+    let config = gfx::SwapchainConfig::new()
+                    .with_color::<ColorFormat>();
+    let mut swap_chain = surface.build_swapchain(config, &graphics_queue);
+
+    let views: Vec<gfx::handle::RenderTargetView<device_gl::Resources, ColorFormat>> =
+        swap_chain
+            .get_backbuffers()
+            .iter()
+            .map(|&(ref color, ref ds)| {
+                let color_desc = texture::RenderDesc {
+                    channel: ColorFormat::get_format().1,
+                    level: 0,
+                    layer: None,
+                };
+                let rtv = factory.view_texture_as_render_target_raw(color, color_desc)
```

Comment:
The color format in `SwapchainConfig` is currently set on runtime, so that we could query for support the surface in the future in case the prefered format might not be supported.